Sources/RESP3/RESP3Token.swift

```diff
@@ -15,6 +15,162 @@
 import NIOCore
 
 public struct RESP3Token: Hashable, Sendable {
+    public struct Unchecked: Hashable, Sendable {
```

Comment:
What's the benefit of this struct? And why is it public?

Reply:
The main issue in the benchmarks was that we verify the data recursively when initializing the type, and then need to do a _second_ pass to go through the results. While our parsing is faster in RESP3 compared to the old RESP implementation, doing it twice made it heavier.

The Unchecked type ensures we can capture data without checking it _on initialization_, defering that operation until the value is accessed. The regular RESP3 token maintains the old behaviour, which I think is helpful for end users. However, since we also have `RESP3Token.Value`, we'll need to see what we'll want to end up exposing in public APIs such as `.get` and pub/sub.